### PR TITLE
Rx rewrite of repos

### PR DESCRIPTION
**Pull request by kmaida**
I have rewritten the Angular sample repo to use RxJS, an HTTP interceptor, and have performed some cleanup. The samples now support and align with QS tutorial updates in this PR: https://github.com/auth0/docs/pull/8009

### CODE REVIEW

File: 01-Login/.vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (luisrudge) [inline 01-Login/.vscode/settings.json]: remove this file

File: 01-Login/src/app/auth/auth.service.ts [hunk 2]
```diff
@@ -2,42 +2,127 @@ import { Injectable } from '@angular/core';
...
+    catchError(err => throwError(err))
+  );
+  // Define observables for SDK methods that return promises by default
+  // For each Auth0 SDK method, first ensure the client instance is ready
+  // concatMap: Using the client instance, call SDK method; SDK returns a promise
+  // from: Convert that resulting promise into an observable
+  isAuthenticated$ = this.auth0Client$.pipe(
+    concatMap((client: Auth0Client) => from(client.isAuthenticated()))
+  );
+  getUser$ = this.auth0Client$.pipe(
```
Comment (luisrudge) [inline 01-Login/src/app/auth/auth.service.ts]: will this forward all the params to `getUser`?

File: 01-Login/src/app/auth/auth.service.ts [hunk 3]
```diff
@@ -2,42 +2,127 @@ import { Injectable } from '@angular/core';
...
+  loggedIn: boolean = null;
 
-  private auth0Client: Auth0Client;
+  constructor(private router: Router) { }
 
-  config = config;
+  checkAuthOnInit() {
+    // This should only be called on app initialization
+    // Check if user already has an active session with Auth0
+    const checkAuth$ = this.isAuthenticated$.pipe(
```
Comment (luisrudge) [inline 01-Login/src/app/auth/auth.service.ts]: why do you need this during init? after `createAuth0Client`, you already know if you're authenticated or not, so I'm not sure why would you need to run this again.

File: 01-Login/src/app/app-routing.module.ts [hunk 4]
```diff
@@ -20,14 +16,14 @@ export const routes: Routes = [
     canActivate: [AuthGuard]
   },
   {
-    path: 'login',
```
Comment (luisrudge) [inline 01-Login/src/app/app-routing.module.ts]: why we don't have this path anymore?

Comment (kmaida) [reply 01-Login/src/app/app-routing.module.ts]: This path wasn't being used for anything / wasn't called anywhere in the app, as far as I could tell. It didn't have a component associated with it and was never requested

File: 01-Login/src/app/auth/auth.service.ts (hunk 3 above)
Comment (kmaida) [reply 01-Login/src/app/auth/auth.service.ts]: Wasn't aware of that feature of the SDK! I will update to remove it. Thanks!

File: 01-Login/.vscode/settings.json (hunk 1 above)
Comment (kmaida) [reply 01-Login/.vscode/settings.json]: It was in the .gitignore, somehow it wasn't ignored. I'll remove it!

**Comment by luisrudge** [comment]
will be merged in https://github.com/auth0-samples/auth0-angular-samples/pull/167

**Pull request closed without merging**